app/javascript/crayons/MarkdownToolbar/MarkdownToolbar.jsx

```diff
@@ -181,12 +182,36 @@ export const MarkdownToolbar = ({ textAreaId }) => {
   const insertSyntax = (syntaxName) => {
     setOverflowMenuOpen(false);
 
-    const { newTextAreaValue, newCursorStart, newCursorEnd } =
-      markdownSyntaxFormatters[syntaxName].getFormatting(textArea);
+    const {
+      newCursorStart,
+      newCursorEnd,
+      editSelectionStart,
+      editSelectionEnd,
+      replaceSelectionWith,
```

Comment:
The `execCommand('insertText')` function overwrites just a given section of the text area, so rather than replace the full text area value, we specify the area we're editing, and overwrite it with the `replaceSelectionWith` text.

If we used the `execCommand` function to overwrite the whole text area (similar to before), when a user uses `cmd + z` to undo, the whole text area would end up highlighted, rather than the appropriate section